rocAL/source/pipeline/master_graph.cpp

```diff
@@ -1136,7 +1136,8 @@ std::vector<rocalTensorList *> MasterGraph::create_label_reader(const char *sour
         THROW("A metadata reader has already been created")
     if (_augmented_meta_data)
         THROW("Metadata can only have a single output")
-
+    if (strlen(source_path) == 0)
```

Comment:
```suggestion
    if (strlen(source_path) == 0)
```
use is_empty instead

Reply:
The source path is passed as const char array so strlen is used here

Reply:
can't you check for nullptr in that case?